contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -25,7 +25,6 @@ root="${root##zfs:}"
 root="${root##ZFS=}"
 
 [ -d "$GENERATOR_DIR" ] || mkdir "$GENERATOR_DIR"
-[ -d "$GENERATOR_DIR/sysroot.mount.d" ] || mkdir "$GENERATOR_DIR/sysroot.mount.d"
 
 {
```

Comment:
This generator-written unit is incomplete.

The enhancement.d pattern is better because we are out of tree and we do not benefit from upstream changes or rewrites, so we *want* to pick up whatever attributes the original sysroot.mount unit had.  For example: in Fedora 23, "Before=sysroot.mount" is defined in base-generated unit, but in older versions, it might be a different dependency set, and in newer versions, it may change (and that change we do not control, but we need to be compatible with it).